src/m3ninx/index/segment/mem/terms_dict.go

```diff
@@ -84,6 +85,27 @@ func (d *termsDict) Fields() sgmt.FieldsIterator {
 	return newBytesSliceIter(fields, d.opts)
 }
 
+func (d *termsDict) FieldsPostingsList() sgmt.FieldsPostingsListIterator {
+	d.fields.RLock()
+	defer d.fields.RUnlock()
+	// NB(bodu): This is probably fine since the terms dict/mem segment is only used in tests.
+	fields := make([]uniqueField, 0, d.fields.Len())
+	for _, entry := range d.fields.Iter() {
+		field := entry.Key()
+		pl := roaring.NewPostingsList()
+		if postingsMap, ok := d.fields.Get(field); ok {
+			for _, entry := range postingsMap.Iter() {
+				pl.Union(entry.value)
+			}
```

Comment:
Even though only used in tests, might be worth just doing at least a `pl.UnionMany(...)`?  Just since it'll be O(1) vs O(n) and all we need to do is have an array here to collect them and call it once.